Add the uid-/gidnumber as an argument to ipa_user

smartsammler opened the pull request:
<!--- Describe the change, including rationale and design decisions -->
##### SUMMARY
Fixes issue #32555

* Adds the uidnumber and the gidnumber which are called UID and GID in IPA's web interface, but are called `uidnumber` and `gidnumber` in IPA's JSON API and therefore in ansible's `ipa_user`s return value.
* To avoid confusion with `ipa_user`'s `uid` I stayed with the JSON API names of the `user_mod` command.
* I changed the code analogue to PR #32369

<!--- Describe the change, including rationale and design decisions -->

<!---
If you are fixing an existing issue, please include "Fixes #nnn" in your
commit message and your description; but you should still explain what
the change does.
-->

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - Feature Pull Request

##### COMPONENT NAME
<!--- Name of the module/plugin/module/task -->
ipa_user

##### ANSIBLE VERSION
<!--- Paste verbatim output from "ansible --version" between quotes below -->
```
N/A
```


##### ADDITIONAL INFORMATION
I tested it locally with the following setup:
IPA client:
--------------
Fedora 25 workstation 
```
$ ipa --version
VERSION: 4.4.4, API_VERSION: 2.215
```

IPA server: 
----------------
CentOS 7
```
$ ipa --version
 VERSION: 4.5.0, API_VERSION: 2.228
```

ansible role
------------------
``` yaml
---
- name: activate user
  ipa_user:
    ipa_host: my_ipa_server.tld
    ipa_pass: top_secret
    uid: testuser
    state: enabled
    uidnumber: 1030
    gidnumber: 1030
```

<!---
Include additional information to help people understand the change here.
For bugs that don't have a linked bug report, a step-by-step reproduction
of the problem is helpful.
  -->

<!--- Paste verbatim command output below, e.g. before and after your change -->

smartsammler committed:
Adds the uid-/gidnumber as an argument to ipa_user

* Fixes issue #32555
* Adds the uidnumber which is called UID in the IPA web interface most
often, but called uidnumber in the JSON-API of IPA's `user_mod`
call.
* Adds the gidnumber which is called GID in the IPA web interface, but
called gidnumber in the JSON-API
* Code changes analogue to PR #32369

Akasurde commented on lib/ansible/modules/identity/ipa/ipa_user.py:
```diff
@@ -49,12 +49,21 @@
     - List of telephone numbers assigned to the user.
     - If an empty list is passed all assigned telephone numbers will be deleted.
     - If None is passed telephone numbers will not be checked or changed.
+    version_added: 2.5
```
Telephonenumber is already added in version 2.3.

smartsammler replied:
Uups. Sorry, should be fixed now.

Akasurde reviewed (changes requested):
Rest LGTM

smartsammler committed:
Removed wrong version_number from telephonenumber

Remove wrongly placed version_number information

Akasurde reviewed (approved):
shipit

resmo commented on lib/ansible/modules/identity/ipa/ipa_user.py:
```diff
@@ -217,7 +231,8 @@ def ensure(module, client):
                                 telephonenumber=module.params['telephonenumber'], title=module.params['title'],
-                                userpassword=module.params['password'])
+                                userpassword=module.params['password'],
+                                gidnumber=module.params.get('gidnumber'), uidnumber=module.params.get('uidnumber'))
```
would you mind creating 2 lines here? for each param?

resmo commented on lib/ansible/modules/identity/ipa/ipa_user.py:
```diff
@@ -250,6 +265,7 @@ def main():
                          mail=dict(type='list'),
                          sn=dict(type='str'),
                          uid=dict(type='str', required=True, aliases=['name']),
+                         gidnumber=dict(type='str'), uidnumber=dict(type='str'),
```
would you mind creating 2 lines here? for each param?

smartsammler replied:
done

smartsammler committed:
Put arguments in different lines

Following PEP-8 the uidnumber and gidnumber are on different lines in the update method
